ls1intum/Artemis
# [Enhancement] - Improve Detail View Performance

### Checklist

### Motivation and Context


For very big courses, fetching the data can take quite long. People who want to see basic course information need to wait until all the data is fetched.

### Description

We separate fetching the basic course information and the complex statistic data, so we can see the basic information from the beginning on.
So see that we still fetch information, we add spinning icons to each graph

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Management
3. Navigate into a Detail Page by clicking in the course header
4. Make sure that while the data is getting fetched, spinning icons are shown for each graph

### Screenshots


A screencast how it should look:

https://user-images.githubusercontent.com/44401560/117706958-3bd31280-b1ce-11eb-9c22-a3988db093a0.mov



You can already see basic course information while the graphs are loading:
<img width="1410" alt="Screenshot 2021-05-10 at 20 24 18" src="https://user-images.githubusercontent.com/44401560/117706749-f9113a80-b1cd-11eb-9c0d-43d6d8cd8348.png">
